src/lighteval/tasks/requests.py

```diff
@@ -38,6 +44,9 @@ class RequestType(Enum):
     GREEDY_UNTIL_MULTI_TURN = auto()
 
 
+Context: TypeAlias = object
```

Comment:
I introduced this type to account for both `str` and `Conversation` but in the future it could be for example ‍`huggingface_hub.DocumentQuestionAnsweringInputData` for Document Question Answering

Reply:
An idea: currently `task.fewshot_sampler.fewshot_context()` is the ultimate responsible for creating the context for a doc even if the task hasn't a few-shot setting. We could imagine having a  `context_augmenters` attribute for the task, containing everything that could augment the context like a few-shot manager or a RAG retriever and have them one by one apply themselves to the context ,starting from initial context which is the instruction+query, in the task's `create_context()` method.